pkg/api/annotations.go

```diff
@@ -346,3 +349,33 @@ func GetAnnotationTags(c *models.ReqContext) response.Response {
 
 	return response.JSON(200, annotations.GetAnnotationTagsResponse{Result: result})
 }
+
+// AnnotationTypeScopeResolver provides an AttributeScopeResolver able to
+// resolve annotation types. Scope "annotations:id:<id>" will be translated to "annotations:type:<type>,
+// where <type> is the type of annotation with id <id>.
+func AnnotationTypeScopeResolver() (string, accesscontrol.AttributeScopeResolveFunc) {
+	annotationTypeResolver := func(ctx context.Context, orgID int64, initialScope string) (string, error) {
+		scopeParts := strings.Split(initialScope, ":")
+		if scopeParts[0] != accesscontrol.ScopeAnnotationsRoot || len(scopeParts) != 3 {
+			return "", accesscontrol.ErrInvalidScope
+		}
+
+		annotationIdStr := scopeParts[2]
+		annotationId, err := strconv.Atoi(annotationIdStr)
+		if err != nil {
+			return "", accesscontrol.ErrInvalidScope
+		}
+
+		ann, resp := findAnnotationByID(annotations.GetRepository(), int64(annotationId), orgID)
```

Comment:
Yeah, that's a good point. We have a similar issue for other scope resolvers. I think it's fine to leave it like that for now, but we should think about cacheing the result in the future to avoid querying for the same object twice.